File: api/src/main/java/org/delivery/api/domain/storemenu/service/StoreMenuService.java
/*
package org.delivery.api.domain.storemenu.service;

import lombok.RequiredArgsConstructor;
import org.delivery.common.error.ErrorCode;
import org.delivery.common.exception.ApiException;
import org.delivery.db.storemenu.StoreMenuEntity;
import org.delivery.db.storemenu.StoreMenuRepository;
import org.delivery.db.storemenu.enums.StoreMenuStatus;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@RequiredArgsConstructor
@Service
public class StoreMenuService {

    private final StoreMenuRepository storeMenuRepository;

    public StoreMenuEntity getStoreMenuWithThrow(Long id){
        var entity = storeMenuRepository.findFirstByIdAndStatusOrderByIdDesc(id, StoreMenuStatus.REGISTER);

        return Optional.ofNullable(entity) .orElseThrow(() -> new ApiException(ErrorCode.NULL_POINT));
    }

    public List<StoreMenuEntity> getStoreMenuByStoreId(Long storeId){
        return storeMenuRepository.findAllByStoreIdAndStatusOrderBySequenceDesc(storeId, StoreMenuStatus.REGISTER);
    }

    public StoreMenuEntity register(StoreMenuEntity storeMenuEntity){
        return Optional.ofNullable(storeMenuEntity)
                .map(it ->{
                    it.setStatus(StoreMenuStatus.REGISTER);
                    return storeMenuRepository.save(it);
                })
                .orElseThrow(() -> new ApiException(ErrorCode.NULL_POINT));
    }
}
*/

File: api/src/main/kotlin/org/delivery/api/domain/storemenu/service/StoreMenuService.kt
package org.delivery.api.domain.storemenu.service

import org.delivery.common.error.ErrorCode
import org.delivery.common.exception.ApiException
import org.delivery.db.storemenu.StoreMenuEntity
import org.delivery.db.storemenu.StoreMenuRepository
import org.delivery.db.storemenu.enums.StoreMenuStatus
import org.springframework.stereotype.Service

@Service
class StoreMenuService(
    val storeMenuRepository: StoreMenuRepository
) {

    fun getStoreMenuWithThrow(
        id:Long
    ): StoreMenuEntity{
        storeMenuRepository.findFirstByIdAndStatusOrderByIdDesc(id,StoreMenuStatus.REGISTER)
            ?.let{
                return it
            }?: throw ApiException(ErrorCode.NULL_POINT)
    }

    fun getStoreMenuByStoreId(
        storeId:Long
    ):List<StoreMenuEntity> {
        return storeMenuRepository.findAllByStoreIdAndStatusOrderBySequenceDesc(
            storeId,
            StoreMenuStatus.REGISTER
        )
    }


    fun register(
        storeMenuEntity: StoreMenuEntity?
    ): StoreMenuEntity {
        storeMenuEntity?.let{
            it.status = StoreMenuStatus.REGISTER
            return storeMenuRepository.save(it)
        }?: throw ApiException(ErrorCode.NULL_POINT)
    }
}